### PR TITLE
Ensure that @ClientHeaderParam works when the rest client contains a method with a primitive return type

### LINKED ISSUE #27959: OIDC scenarion with authorization using `@ClientHeaderParam` doesn't work after move to reactive
### Describe the bug

I'm trying to migrate testing scenarios we have for quarkus-oidc-client-filter / quarkus-oidc-token-propagation / quarkus-rest-client into their reactive variants.

Changes for the move are here: https://github.com/rsvoboda/quarkus-test-suite/commit/818b36371a7f511062377b869fe34407221537f7

There is failure in one scenario called "LookupAuthorization" with authorization using `@ClientHeaderParam` where the Bearer token gets constructed using direct communication with oidc server.
https://github.com/quarkus-qe/quarkus-test-suite/blob/main/security/keycloak-oidc-client-extended/src/main/java/io/quarkus/ts/security/keycloak/oidcclient/extended/restclient/ping/clients/LookupAuthorizationPongClient.java#L25

I can't say for sure if the trouble is in resteasy-reactive, rest-client-reactive or in oidc extensions.
Thus CCing @sberyozkin, @Sgitario (who created that test coverage ;) ) and @geoand 

### Expected behavior

All tests to pass

### Actual behavior

Test failure
```
[ERROR] Failures:
[ERROR]   OidcRestClientIT.testLookupAuthorization:49->assertPingEndpoints:57->assertPingPong:109 1 expectation failed.
Expected status code <200> but was <500>.

[INFO]
[ERROR] Tests run: 9, Failures: 1, Errors: 0, Skipped: 0
```

### How to Reproduce?

 - clone https://github.com/rsvoboda/quarkus-test-suite
 - checkout `oidc-client-reactive-filter` branch
 - run `mvn clean verify -pl security/keycloak-oidc-client-extended`

### Output of `uname -a` or `ver`

macOS Monterey

### Output of `java -version`

Java 17

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

Quarkus main

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by geoand**
Resolves: #27959

### CODE REVIEW

File: extensions/resteasy-reactive/rest-client-reactive/deployment/src/main/java/io/quarkus/rest/client/reactive/deployment/MicroProfileRestClientEnricher.java [hunk 1]
```diff
@@ -445,7 +433,7 @@ private String mockInterface(ClassInfo declaringClass, BuildProducer<GeneratedCl
                 for (MethodInfo method : methods) {
                     if (Modifier.isAbstract(method.flags())) {
                         MethodCreator methodCreator = classCreator.getMethodCreator(MethodDescriptor.of(method));
-                        methodCreator.returnValue(methodCreator.loadNull());
+                        methodCreator.throwException(IllegalStateException.class, "This should never be called");
```
Comment (Sgitario) [inline extensions/resteasy-reactive/rest-client-reactive/deployment/src/main/java/io/quarkus/rest/client/reactive/deployment/MicroProfileRestClientEnricher.java:436]: so, the issue was that we were returning null for primitive types.

**Review by Sgitario — APPROVED** [review]
lgtm!

**Comment by geoand** [comment]
The test failures definitely look related

**Comment by sberyozkin** [comment]
`oidc-token-propagation-reactive` is likely not, I've seen it failing here https://github.com/quarkusio/quarkus/pull/27737#issuecomment-1237440403, I'll add to the list in the OIDC flaky tests issue

**Comment by geoand** [comment]
Yeah, I meant the TCK tests

**Review by gastaldi — APPROVED** [review]

**Pull request merged**